pages/portfolio/index.tsx

```diff
@@ -7,29 +7,42 @@ import InsurancePortfolio from '@components/Portfolio/Insurance';
 import { LeftPanel, RightPanel } from '@components/Portfolio';
 import Footer from '@components/Footer';
 import { FactoryContext, initialFactoryState } from '@context/FactoryContext';
+import Tracer from '@libs/Tracer';
+import { InsuranceStore } from '@context/InsuranceContext';
 
 export default styled(({ className }) => {
     const [tab, setTab] = useState(0);
     const tabs = ['Trading Portfolio', 'Insurance Portfolio'];
     const { allFilledOrders, factoryState: { tracers } = initialFactoryState } = useContext(FactoryContext);
+    const [fetchedTracers, setFetchedTracers] = useState<Tracer[]>([]);
+
+    useEffect(() => {
+        setTimeout(() => {
+            setFetchedTracers(Object.values(tracers));
+        }, 1000);
+    }, [tracers]);
```

Comment:
This is a race condition against fetching the data. So assumes that all the correct values are fetched within 1 second and then loads in the tracers. In general its just not very nice code since it is not deterministic. Am pretty opposed to letting it in, look into where how and why the data is getting loaded in and then if its still to difficult then I can have a look at it.